driver/pace/driver/run.py

```diff
@@ -64,6 +71,11 @@ def get_driver_state(
         metric_terms = pace.util.grid.MetricTerms(
             quantity_factory=quantity_factory, communicator=communicator
         )
+        grid_data = pace.util.grid.GridData.new_from_metric_terms(metric_terms)
+        damping_coeffient = DampingCoefficients.new_from_metric_terms(metric_terms)
+        driver_grid_data = pace.util.grid.DriverGridData.new_from_metric_terms(
+            metric_terms
```

Comment:
I think we need to consolidate `GridData` and `DriverGridData`, mostly just rolling whatever in driver grid data to grid data. There're also some additional changes we need to add in fortran grid savepoint, currently the variables in `DriverGridData` are not in that savepoint.

Reply:
I thought the direction we wanted to go was distinct DriverGridData, PhysicsGridData, and GridData(which could be named DycoreGridData)? You'd rather have one large bucket that has all 3 and contains irrelevant variables when running dycore only or physics only?

Reply:
I read what @elynnwu wrote as being that we can have a `GridData` class with `grid_data.dycore`, `grid_data.physics` and `grid_data.<whatever we call the conversion part currently referred to as DriverGridData>`. Keeps the convenience of one place to initialize or access the data at the top level, but also the nice interfaces at lower levels where you only need to create relevant variables.

I'm more open on this one versus the configuration classes though, since grid data can be generated with very few free parameters/config options. One bin can be OK as long as there are very few free parameters needed to initialize it.

Reply:
This is a good point. I hadn't thought about the case where a user just wants to run one component when I made the comment. I think we will have to go through the variables one by one to see which way makes more sense, updating the jira card now.